compiler/natives/src/runtime/runtime.go

```diff
@@ -172,15 +211,39 @@ func SetFinalizer(x, f interface{}) {
 }
 
 type Func struct {
+	name string
+	file string
+	line int
+
 	opaque struct{} // unexported field to disallow conversions
 }
 
-func (_ *Func) Entry() uintptr                              { return 0 }
-func (_ *Func) FileLine(pc uintptr) (file string, line int) { return "", 0 }
-func (_ *Func) Name() string                                { return "" }
+func (_ *Func) Entry() uintptr { return 0 }
+
+func (f *Func) FileLine(pc uintptr) (file string, line int) {
+	if f == nil {
+		return "", 0
+	}
+	return f.file, f.line
+}
+func (f *Func) Name() string {
+	if f == nil || f.name == "" {
+		return "<unknown>"
+	}
+	return f.name
+}
 
 func FuncForPC(pc uintptr) *Func {
-	return nil
+	ipc := int(pc)
+	if ipc >= len(positionCounters) {
+		// Since we are faking position counters, the only valida way to obtain one
+		// is through a Caller() or Callers() function. If pc is out of positionCounters
+		// bounds it must have been obtained in some other way, which is unexpected.
+		// If a panic proves problematic, we can return a nil *Func, which will
+		// present itself as a generic "unknown" function.
+		panic("GopherJS: pc=" + itoa(ipc) + " is out of range of known position counters")
```

Comment:
If my reading of upstream is correct, panicking is what they do, too, so I expect this is fine.

https://golang.org/src/runtime/symtab.go?s=18483:18515#L698

Reply:
I think this is different. In the upstream code they panic if there is no entry in some kind of data table, which I think is compiler-generated. In our case we panic if we try to look up info about a position that wasn't returned by `runtime.Caller()`.